dref/views.py

```diff
@@ -59,8 +60,25 @@ def get_queryset(self):
         )
         if user.is_superuser:
             return queryset
-        else:
-            return Dref.get_for(user)
+        elif not user.is_superuser:
+            # get current user dref regions
+            regions = [0, 1, 2, 3, 4]
+            dref_list = []
+            for region in regions:
+                codename = f"dref_region_admin_{region}"
+                if Permission.objects.filter(group__user=user, codename=codename).exists():
+                    _queryset = queryset.filter(country__region=region).distinct()
+                    dref_list.append(_queryset)
+            drefs = []
+            for dref in dref_list:
+                ids = list(dref.values_list('id', flat=True))
+                for id in ids:
+                    new_dref = Dref.objects.filter(id=id).first()
+                    drefs.append(new_dref.id)
+            if len(drefs):
+                return queryset.filter(models.Q(id__in=drefs) | models.Q(created_by=user)).distinct()
+            else:
+                return Dref.get_for(user)
```

Comment:
This has lots of N+1 queries, Can we do something like this instead?

```suggestion
            dref_admin_regions_id = [
                codename.replace('dref_region_admin_', '')
                for codename in Permission.objects.filter(
                    group__user=user,
                    codename__startswith='dref_region_admin_',
                    # Maybe add additional filter to make sure dref_region_admin_<number> are only returned
                ).values_list('codename', flat=True)
            ]
            if len(dref_admin_regions_id):
                return queryset.filter(
                    models.Q(created_by=user)
                    | models.Q(country__region__in=dref_admin_regions_id)
                ).distinct()
            else:
                return queryset.model.get_for(user)
```